### PR TITLE
feature(wootric): Added User properties patching

### PR DESCRIPTION
**Pull request by Gauravudia**
## Description of the change

> Added user properties patching implementation along with code refector , test case update

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/wootric/util.js [hunk 1]
```diff
@@ -86,12 +102,12 @@ const retrieveUserId = async (userId, accessToken) => {
...
 
   if (processedUserResponse.status === 200) {
-    return processedUserResponse.response?.id;
+    return processedUserResponse.response;
   }
 
   if (processedUserResponse.status !== 404) {
     throw new CustomError(
-      `[Wootric]:: Unable to retrieve userid due to ${JSON.stringify(
+      `[Wootric]:: Unable to retrieve userId due to ${JSON.stringify(
```
Comment (utsabc) [inline v0/destinations/wootric/util.js:110]: In 404 situation we are not handing the case outside

File: v0/destinations/wootric/util.js [hunk 2]
```diff
@@ -216,106 +247,50 @@ const createDeclinePayloadBuilder = message => {
...
 
 /**
  * Formats identify payload
- * ref: https://docs.wootric.com/api/#get-a-specific-end-user-by-phone-number
  * @param {*} payload
  */
 const formatIdentifyPayload = payload => {
-  if (payload.last_surveyed) {
-    payload.last_surveyed = `${payload.last_surveyed}`;
+  const rawPayload = { ...payload };
```
Comment (utsabc) [inline v0/destinations/wootric/util.js:264]: Lets add bit more contextual documentation for this function. Also the this function is stringifying dates but not formatting anything else for the entire payload. Contextual naming would be helpful?

File: v0/destinations/wootric/transform.js [hunk 3]
```diff
@@ -50,11 +49,14 @@ const identifyResponseBuilder = async (message, destination) => {
...
-    wootricEndUserId = await retrieveUserId(userId, accessToken);
-  }
+  const rawEndUserId = getDestinationExternalID(message, "wootricEndUserId");
+  const userId = getFieldValueFromMessage(message, "userIdOnly");
+  const userDetails = await retrieveUserDetails(
+    rawEndUserId,
+    userId,
+    accessToken
+  );
+  const wootricEndUserId = userDetails?.id;
```
Comment (utsabc) [inline v0/destinations/wootric/transform.js:59]: In 404 situation how would this work

File: v0/destinations/wootric/util.js (hunk 1 above)
Comment (Gauravudia) [reply v0/destinations/wootric/util.js:109]: updated

File: v0/destinations/wootric/util.js (hunk 2 above)
Comment (Gauravudia) [reply v0/destinations/wootric/util.js:266]: updated

File: v0/destinations/wootric/transform.js (hunk 3 above)
Comment (Gauravudia) [reply v0/destinations/wootric/transform.js:59]: updated

**Review by utsabc — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Pull request merged**